Those selenium tests seem real, but they don't happen locally ...
```
FAILED lib/galaxy_test/selenium/test_workflow_run.py::WorkflowRunTestCase::test_execution_with_custom_invocation_report
FAILED lib/galaxy_test/selenium/test_workflow_run.py::WorkflowRunTestCase::test_execution_with_custom_invocation_report
```

--- a/client/src/components/Datatypes/factory.js
+++ b/client/src/components/Datatypes/factory.js
@@ -1,7 +1,7 @@
 import { DatatypesMapperModel } from "./model.js";
 import { getDatatypes } from "./services";
 
-export async function getDatatypesMapper() {
-    const typesAndMapping = await getDatatypes();
+export async function getDatatypesMapper(upload_only = true) {
+    const typesAndMapping = await getDatatypes(upload_only);
     return new DatatypesMapperModel(typesAndMapping);
 }

--- a/client/src/components/Datatypes/model.js
+++ b/client/src/components/Datatypes/model.js
@@ -1,6 +1,7 @@
 export class DatatypesMapperModel {
     constructor(typesAndMapping) {
         this.datatypes = typesAndMapping.datatypes;
+        this.datatypes.sort();
         this.datatypesMapping = typesAndMapping.datatypes_mapping;
     }
 

--- a/client/src/components/Datatypes/services.js
+++ b/client/src/components/Datatypes/services.js
@@ -2,9 +2,13 @@ import axios from "axios";
 import { rethrowSimple } from "utils/simple-error";
 import { getAppRoot } from "onload/loadConfig";
 
-export async function getDatatypes() {
+export async function getDatatypes(upload_only = true) {
     try {
-        const request = await axios.get(`${getAppRoot()}api/datatypes/types_and_mapping`);
+        const request = await axios.get(`${getAppRoot()}api/datatypes/types_and_mapping`, {
+            params: {
+                upload_only: upload_only,
+            },
+        });
         return request.data;
     } catch (e) {
         rethrowSimple(e);

--- a/client/src/components/Markdown/Elements/HistoryDatasetDisplay.vue
+++ b/client/src/components/Markdown/Elements/HistoryDatasetDisplay.vue
@@ -33,44 +33,42 @@
         </b-card-header>
         <b-card-body>
             <UrlDataProvider :url="itemUrl" v-slot="{ result: itemContent, loading, error }">
-                <LoadingSpan v-if="loading" message="Loading Dataset" />
-                <div v-else-if="error">{{ error }}</div>
-                <div v-else class="embedded-dataset content-height">
-                    <UrlDataProvider :url="datatypesUrl" v-slot="{ result: datatypesModel, loading: datatypesLoading }">
-                        <LoadingSpan v-if="datatypesLoading" message="Loading Datatypes" />
-                        <div v-else>
-                            <b-embed
-                                v-if="isSubTypeOfAny(datasetType, ['pdf', 'html'], datatypesModel)"
-                                type="iframe"
-                                aspect="16by9"
-                                :src="displayUrl"
-                            />
-                            <div v-else-if="itemContent.item_data">
-                                <div v-if="isSubTypeOfAny(datasetType, ['tabular'], datatypesModel)">
-                                    <UrlDataProvider
-                                        :url="metaUrl"
-                                        v-slot="{ result: metaData, loading: metaLoading, error: metaError }"
-                                    >
-                                        <LoadingSpan v-if="metaLoading" message="Loading Metadata" />
-                                        <div v-else-if="metaError">{{ metaError }}</div>
-                                        <b-table
-                                            v-else
-                                            striped
-                                            hover
-                                            :fields="getFields(metaData)"
-                                            :items="getItems(itemContent.item_data, metaData)"
-                                        />
-                                    </UrlDataProvider>
-                                </div>
-                                <pre v-else>
+                <UrlDataProvider :url="datatypesUrl" v-slot="{ result: datatypesModel, loading: datatypesLoading }">
+                    <LoadingSpan v-if="loading" message="Loading Dataset" />
+                    <LoadingSpan v-else-if="datatypesLoading" message="Loading Datatypes" />
+                    <div v-else-if="error">{{ error }}</div>
+                    <div v-else class="embedded-dataset content-height">
+                        <b-embed
+                            v-if="isSubTypeOfAny(datasetType, ['pdf', 'html'], datatypesModel)"
+                            type="iframe"
+                            aspect="16by9"
+                            :src="displayUrl"
+                        />
+                        <div v-else-if="itemContent.item_data">
+                            <div v-if="isSubTypeOfAny(datasetType, ['tabular'], datatypesModel)">
+                                <UrlDataProvider
+                                    :url="metaUrl"
+                                    v-slot="{ result: metaData, loading: metaLoading, error: metaError }"
+                                >
+                                    <LoadingSpan v-if="metaLoading" message="Loading Metadata" />
+                                    <div v-else-if="metaError">{{ metaError }}</div>
+                                    <b-table
+                                        v-else
+                                        striped
+                                        hover
+                                        :fields="getFields(metaData)"
+                                        :items="getItems(itemContent.item_data, metaData)"
+                                    />
+                                </UrlDataProvider>
+                            </div>
+                            <pre v-else>
                                     <code class="text-normalwrap">{{ itemContent.item_data }}</code>
                                 </pre>
-                            </div>
-                            <div v-else>No content found.</div>
-                            <b-link v-if="itemContent.truncated" :href="itemContent.item_url"> Show More... </b-link>
                         </div>
-                    </UrlDataProvider>
-                </div>
+                        <div v-else>No content found.</div>
+                        <b-link v-if="itemContent.truncated" :href="itemContent.item_url"> Show More... </b-link>
+                    </div>
+                </UrlDataProvider>
             </UrlDataProvider>
         </b-card-body>
     </b-card>

--- a/client/src/components/ToolRecommendation.vue
+++ b/client/src/components/ToolRecommendation.vue
@@ -55,7 +55,7 @@ export default {
                 tool_sequence: toolId,
             };
             getToolPredictions(requestData).then((responsePred) => {
-                getDatatypesMapper().then((datatypesMapper) => {
+                getDatatypesMapper(false).then((datatypesMapper) => {
                     const predData = responsePred.predicted_data;
                     this.deprecated = predData.is_deprecated;
                     this.deprecatedMessage = predData.message;

--- a/client/src/components/Workflow/Editor/Index.vue
+++ b/client/src/components/Workflow/Editor/Index.vue
@@ -289,7 +289,7 @@ export default {
     },
     created() {
         this.lastQueue = new LastQueue();
-        getDatatypesMapper().then((mapper) => {
+        getDatatypesMapper(false).then((mapper) => {
             this.datatypesMapper = mapper;
             this.datatypes = mapper.datatypes;
 

--- a/lib/galaxy/selenium/navigation.yml
+++ b/lib/galaxy/selenium/navigation.yml
@@ -579,6 +579,10 @@ workflow_editor:
       type: xpath
       selector: >
         //div[@data-label='Change datatype' and not(ancestor::div[contains(@style,'display: none')])]//span[contains(@class, 'select2-chosen')]
+    select_dataype_text_search:
+      type: xpath
+      selector: >
+        //div[@class="select2-search" and not(ancestor::div[contains(@style,'display: none')])]//input
     select_datatype:
       type: xpath
       selector: >

--- a/lib/galaxy_test/selenium/test_workflow_editor.py
+++ b/lib/galaxy_test/selenium/test_workflow_editor.py
@@ -530,6 +530,7 @@ steps:
         self.set_text_element(output_label, 'workflow output label')
         self.set_text_element(editor.rename_output, 'renamed_output')
         editor.change_datatype.wait_for_and_click()
+        editor.select_dataype_text_search.wait_for_and_send_keys('bam')
         editor.select_datatype(datatype='bam').wait_for_and_click()
         self.set_text_element(editor.add_tags, '#crazynewtag')
         self.set_text_element(editor.remove_tags, '#oldboringtag')

--- a/lib/galaxy_test/selenium/test_workflow_editor.py
+++ b/lib/galaxy_test/selenium/test_workflow_editor.py
@@ -297,6 +297,19 @@ steps:
         self.workflow_editor_connect("text_input_step#out_file1", "collection_input#input1")
         self.assert_connected("text_input_step#out_file1", "collection_input#input1")
 
+    def test_connecting_display_in_upload_false_connections(self):
+        self.open_in_workflow_editor("""
+class: GalaxyWorkflow
+steps:
+  step1:
+    tool_id: test_sam_to_bam_conversions
+  step2:
+    tool_id: test_sam_to_bam_conversions
+        """)
+
+        self.workflow_editor_connect("step1#qname_input_sorted_bam_output", "step2#input5")
+        self.assert_connected("step1#qname_input_sorted_bam_output", "step2#input5")
+
     @selenium_test
     def test_existing_connections(self):
         self.open_in_workflow_editor(WORKFLOW_SIMPLE_CAT_TWICE)

